### PR TITLE
Fix libgit2 build scripts

### PR DESCRIPTION
**Pull request by phatblat**
I had some trouble getting libgit2 to build in 0.5 and 0.6, getting errors about `'libssh2.h' file not found`. I wasn't sure if it was my setup (been on Xcode and Yosemite betas a lot lately), but by modifying the update_libgit2 scripts and defining `LIBSSH2_INCLUDE_DIR` it builds consistently.

Addresses #442

### CODE REVIEW

File: script/update_libgit2 [hunk 1]
```diff
@@ -22,7 +22,11 @@ fi
 mkdir build
 cd build
 
-cmake -DBUILD_SHARED_LIBS:BOOL=OFF -DBUILD_CLAR:BOOL=OFF -DTHREADSAFE:BOOL=ON ..
+cmake -DBUILD_SHARED_LIBS:BOOL=OFF \
+	-DLIBSSH2_INCLUDE_DIR:PATH=../../External/libssh2-ios/include/libssh2/ \
```
Comment (jspahrsummers) [inline script/update_libgit2]: This won't be correct for OS X builds, which use the system libssh2 (IIRC).

**Comment by jspahrsummers** [comment]
Thanks! The iOS changes definitely look good, but I'm a bit unsure about the OS X path.

**Comment by phatblat** [comment]
Went ahead and reverted the change to the Mac build script. When I run `script/update_libgit2` from the command line it builds fine.

It's only when I build the libgit2 target in Xcode that it fails.

```
[ 88%] Building C object CMakeFiles/git2.dir/src/transports/smart.c.o
[ 89%] Building C object CMakeFiles/git2.dir/src/transports/smart_pkt.c.o
[ 90%] Building C object CMakeFiles/git2.dir/src/transports/smart_protocol.c.o
[ 91%] Building C object CMakeFiles/git2.dir/src/transports/ssh.c.o
/Users/phatblat/dev/ios/APP_NAME/Carthage/Checkouts/objective-git/External/libgit2/src/transports/ssh.c:9:10: fatal error: 'libssh2.h' file not found
#include <libssh2.h>
         ^
1 error generated.
make[2]: *** [CMakeFiles/git2.dir/src/transports/ssh.c.o] Error 1
make[1]: *** [CMakeFiles/git2.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by pietbrauer** [comment]
For me both (Mac and iOS) built when I checked out @phatblat branch. Just using the iOS changes didn't solve the building issues.

**Comment by jspahrsummers** [comment]
Sorry, to clear: I think `LIBSSH2_INCLUDE_DIR` is important to pass to the Mac target, I just don't think it should point into the iOS build of libssh2.

**Comment by phatblat** [comment]
`/usr/local/include` works

**Comment by jspahrsummers** [comment]
Awesome, thanks for tackling this! :sparkles:

**Pull request merged**